lib/puppet/provider/cisco_interface/cisco.rb

```diff
@@ -185,19 +186,17 @@ def self.properties_get(interface_name, nu_obj, all_intf: nil)
     new(current_state)
   end # self.properties_get
 
-  def self.instances(single_intf=nil)
+  def self.instances(single_intf=nil, interface_threshold=0)
```

Comment:
Do you really need the default arg value for `interface_threshold` since you have `facts['interface_threshold'] = 0` in `self.interface_threshold`?

Reply:
Yes and no. `instances` may be called directly via `puppet resource` so it's best to initialize it here (gotta protect my api's ya know), but the way the logic works `interface_threshold` should only be referenced when `single_intf` is also defined. So really it's a best practice to initialize it here.